R/LSTconfidenceIntervals.R

```diff
@@ -0,0 +1,407 @@
+#
+# Copyright (C) 2013-2018 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ConfidenceIntervals <- function(jaspResults, dataset = NULL, options) {
+  
+  confidenceContainer <- .getConfidenceContainer(jaspResults)
+  
+  .simulateDatasets(confidenceContainer, options)
+  
+  .computeConfidenceIntervals(confidenceContainer, options)
+  
+  .makeSummaryTable(confidenceContainer, options)
+  
+  .plotTreePlot(confidenceContainer, options)
+  
+  .makeTreeTable(confidenceContainer, options)
+  
+  .plotDatasets(confidenceContainer, options)
+  
+  .plotConvergencePlot(confidenceContainer, options)
+  
+  return()
+}
+
+.getConfidenceContainer <- function(jaspResults) {
+  if (!is.null(jaspResults[["confidenceContainer"]])) {
+    confidenceContainer <- jaspResults[["confidenceContainer"]]
+  } else {
+    confidenceContainer <- createJaspContainer()
+    # we set the dependencies on the container, this means that all items inside the container automatically have these dependencies
+    confidenceContainer$dependOn(c("mu", "sigma", "n", "confidenceIntervalInterval", "nReps", 
+                                   "treePlot"))
+    jaspResults[["confidenceContainer"]] <- confidenceContainer
+  }
```

Comment:
```suggestion
  confidenceContainer <- jaspResults[["confidenceContainer"]] %setOrRetrieve% createJaspContainer(dependencies = c("mu", "sigma", "n", "confidenceIntervalInterval", "nReps", "treePlot"))
```
requires importing `%setOrRetrieve%` from jaspBase (preferably using roxygen2).